contracts/upgradeable_contracts/arbitrary_message/HomeAMB.sol

```diff
@@ -9,15 +9,15 @@ contract HomeAMB is BasicHomeAMB {
 
     function requireToPassMessage(address _contract, bytes _data, uint256 _gas) public {
         require(keccak256(homeToForeignMode()) == keccak256(SUBSIDIZED_MODE));
-        checkAndUpdateGasLimits(_gas);
+        require(_gas <= maxPerTx());
```

Comment:
I would recommend to extend this check in the way described here: https://github.com/poanetwork/poa-bridge-contracts/issues/73#issuecomment-419936405